sdk/core/System.ClientModel/src/Internal/Logging/ClientModelLogMessages.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace System.ClientModel.Internal;
+
+// The methods in this class should only ever be called from LoggingHandler
+internal partial class ClientModelLogMessages
+{
+    private ILogger _logger;
+    private PipelineMessageSanitizer _sanitizer;
+
+    public ClientModelLogMessages(ILogger logger, PipelineMessageSanitizer sanitizer)
+    {
+        _logger = logger;
+        _sanitizer = sanitizer;
+    }
+
+    #region Request
+
+    public void Request(string requestId, PipelineRequest request, string? clientAssembly)
```

Comment:
Curious about this -- why are there two `Request` methods on this type?

Reply:
For perf, the public method checks if the logger is enabled before doing work like sanitizing/formatting things. If you look at some of the other logs, for example ResponseDelay, there is no extra work needed, so no need to have two methods.

Reply:
The source generated methods include the check by default before writing the log, so that's why there's a SkipEnabledCheck = true in the LoggerMessageAttribute on the private message, since I'm checking that myself in the public method.